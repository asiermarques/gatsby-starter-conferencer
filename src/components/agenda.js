import React from "react"
import agendaMapper from "../lib/agendaMapper"
import agendaSlotType from "../lib/agendaSlotTypes";

export default ({agenda, speakers}) => {
    const tracks = agenda.tracks;
    const agendaSlotMap = agendaMapper(agenda, speakers);
    
    return <section id="agenda">
        <div className="container">
            <h2 className="text-center">Agenda</h2>
            <div className="table-responsive-sm">
            <table id="agenda-table" className="table table-striped stacktable large-only">
                    <thead>
                    <tr>
                        <th></th>
                        {tracks.map( (track, index) => <th key={index} width="40%">{track.name}</th> )}
                    </tr>
                    </thead>
                    <tbody>
                    {agendaSlotMap.map( (slot, slot_index) => { 
                        const [time_range, ...content] = slot;
                        return (<tr key={slot_index}>
                            <td class="time">{time_range}</td>
                            {content.map( (content_item, index) => {
                                
                                if(content_item.type === agendaSlotType.SPEAKER) { 
                                    return (
                                        <td key={index}>
                                            <a href={content_item.content.slug + "#talk"}>
                                                <h5>{content_item.content.talk.title}</h5>
                                                <p>{content_item.content.name}</p>
                                            </a>
                                        </td>
                                    )
                                }else{
                                    return (<td key={index}><p>{content_item.content}</p></td>)
                                }
                            })}
                        </tr>
                        )
                    })}
                    </tbody>
            </table>
            </div>
        </div>
    </section>
}

