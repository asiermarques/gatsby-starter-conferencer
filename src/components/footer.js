import React from "react"
import { graphql, StaticQuery } from "gatsby"
import LocationMap from "./locationMap"
import Sponsors from "./sponsors"
import Organizers from "./organizers"
import Nav from "./nav"


export default () => (
    <StaticQuery
        query={graphql`
        query {
            site {
              siteMetadata {
                conference_hashtag
                organizers {
                  name,
                  link,
                  image
                },
                footer_links {
                  link,
                  title
                },
                location {
                  gmaps_iframe_url,
                  addresses {
                    name,
                    line,
                    map_link
                  }
                },
                sponsor_blocks{
                  name,
                  height_em,
                  sponsors {
                      name,
                      link,
                      image
                  }
                }
              }
            }
          }
        `}
        render={ data => {

          const configData = data.site.siteMetadata;

          return (
            <footer>
                <LocationMap location={configData.location}/>
        
                <section id="about">
        
                  <div class="container">
        
                      <div class="row">
                            <div class="col-md-7">
                              <Organizers organizers={configData.organizers}/>
                            </div>
        
                            <div class="col-md-5 text-center">
                              <h4>About</h4>
                              <Nav items={configData.footer_links}></Nav>
                            </div>
        
                      </div>
                  </div>
        
                </section>
        
                <Sponsors sponsorBlocks={configData.sponsor_blocks}/>
        
                <section>
                    <div className="container">
                        <hr></hr>
                        Created with the <a href="https://github.com/asiermarques/gatsby-starter-conferencer">Gatsby Conferencer starter</a>
                    </div>
                </section>
        
                        
        
            </footer>
                )


        }}
    />
)