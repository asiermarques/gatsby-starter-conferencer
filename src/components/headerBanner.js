import React from "react"
import { graphql, StaticQuery } from "gatsby"


export default () => (
    <StaticQuery
        query={graphql`
        query {
            site {
                siteMetadata {
                    conference_name, 
                    conference_claim,
                    header_banner {
                        cta_pre_text,
                        cta_text
                    }
                }
            }
        }
        `}
        render={ data => (
            <div class="jumbotron">
                <div class="container">
                    <h1 class="display-4">{data.site.siteMetadata.conference_name}</h1>
                    <p class="lead">{data.site.siteMetadata.conference_claim}</p>
                    <hr class="my-4"/>
                    <p>{data.site.siteMetadata.header_banner.cta_pre_text}</p>
                    <a class="btn btn-primary btn-lg" href="/dot" role="button">{data.site.siteMetadata.header_banner.cta_text}</a>
                </div>
            </div>
        )}
    />
)