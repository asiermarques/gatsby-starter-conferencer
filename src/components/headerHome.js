import React from "react"

export default ({ data }) => <div class="jumbotron">
        <div class="container text-center">
            <h1 class="display-4">{data.conference_name}</h1>
            <p class="lead">{data.conference_claim}</p>
            <hr class="my-4"/>
            <p>{data.header_banner.cta_pre_text}</p>
            <a class="btn btn-primary btn-lg" href={data.header_banner.cta_url} role="button">{data.header_banner.cta_text}</a>
        </div>
    </div>