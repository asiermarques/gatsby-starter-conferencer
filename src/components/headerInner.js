import React from "react"

export default ({ data }) => <div class="jumbotron">
        <div class="container">
            <div className="row">
                <div className="col-sm-7">
                    <a href="/"><h2 class="display-4">{data.conference_name}</h2></a>
                    <p class="lead">{data.conference_claim}</p>
                </div>
                <div className="col-sm-5 text-center">
                    <p>{data.header_banner.cta_pre_text}</p>
                    <a class="btn btn-primary btn-lg" href={data.header_banner.cta_url} role="button">{data.header_banner.cta_text}</a>
                </div>
            </div> 
        </div>
    </div>
  

