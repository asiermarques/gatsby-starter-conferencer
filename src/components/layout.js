import React from "react"
// eslint-disable-next-line
import Styles from "../scss/custom.scss"

export default ({ children }) => (
  <div>
    {children}
  </div>
)