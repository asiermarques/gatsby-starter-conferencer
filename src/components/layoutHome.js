import React from "react"
import Styles from "../scss/custom.scss"

import Header from "./headerBanner"

export default ({ children }) => (
  <div>
    <Header></Header>
    {children}
  </div>
)

