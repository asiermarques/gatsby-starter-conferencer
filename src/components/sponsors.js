import React from "react"

export default ({ sponsorBlocks }) => {

    const has_sponsors = !!sponsorBlocks.map( block => block.sponsors.length )
                                         .reduce((prev, current) => prev + current, 0)
    
    return <section id="sponsors">
                <div className="container">
                    {has_sponsors && <>    
                        <h2>Our sponsors</h2>

                        {sponsorBlocks.map( (block, index_block) => (
                        <div key={index_block} className="sponsor_block">
                            {block.name && <h3>{block.name}</h3>}
                            {block.sponsors.map( (sponsor, index) => <a key={index}
                                                                        href={sponsor.link} 
                                                                        title={sponsor.name}
                                                                        class="sponsor">
                                    <img src={sponsor.image} alt={sponsor.name} 
                                         style={{
                                            height: block.height_em + 'em',
                                            width: 'auto'
                                            }}/>
                            </a> )}
                        </div>
                        ))}
                    </>}
                </div>
            </section>
}