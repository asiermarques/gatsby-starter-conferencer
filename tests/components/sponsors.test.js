import React from "react"
import renderer from "react-test-renderer"
import {PureSponsors as Sponsors} from "../../src/components/sponsors"
import {WITH_BLOCK_TITLES as SponsorsConfigFixture, WITHOUT_BLOCK_TITLES as SponsorsWOTitleConfigFixture} from "../__fixtures/SponsorsConfig"

describe("Sponsors", () => {
  it("renders correctly", () => {
    const data = {
        site: {
          siteMetadata: {"sponsor_blocks": SponsorsConfigFixture},
        },
      }
    const tree = renderer.create(<Sponsors data={data}/>).toJSON()
    expect(tree).toMatchSnapshot()
  });
  it("renders correctly without any sponsor", () => {
    const data = {
        site: {
          siteMetadata: {"sponsor_blocks": []},
        },
      }
    const tree = renderer.create(<Sponsors data={data}/>).toJSON()
    expect(tree).toMatchSnapshot()
  });
  it("renders correctly without sponsor block titles", () => {
    const data = {
        site: {
          siteMetadata: {"sponsor_blocks": SponsorsWOTitleConfigFixture},
        },
      }
    const tree = renderer.create(<Sponsors data={data}/>).toJSON()
    expect(tree).toMatchSnapshot()
  });
})