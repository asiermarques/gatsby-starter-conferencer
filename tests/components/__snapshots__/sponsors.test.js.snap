// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsors renders correctly 1`] = `
<section
  id="sponsors"
>
  <div
    className="container"
  >
    <h2>
      Our sponsors
    </h2>
    <div
      className="sponsor_block"
    >
      <h3>
        Uranium
      </h3>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Tyrell Corp."
      >
        <img
          alt="Tyrell Corp."
          src="/test1.png"
          style={
            Object {
              "height": "15em",
              "width": "auto",
            }
          }
        />
      </a>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Black Mesa"
      >
        <img
          alt="Black Mesa"
          src="/test2.png"
          style={
            Object {
              "height": "15em",
              "width": "auto",
            }
          }
        />
      </a>
    </div>
    <div
      className="sponsor_block"
    >
      <h3>
        Platinium
      </h3>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="IN Gen"
      >
        <img
          alt="IN Gen"
          src="/test3.png"
          style={
            Object {
              "height": "10em",
              "width": "auto",
            }
          }
        />
      </a>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Umbrella Corp"
      >
        <img
          alt="Umbrella Corp"
          src="/test4.png"
          style={
            Object {
              "height": "10em",
              "width": "auto",
            }
          }
        />
      </a>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Waine Enterprises"
      >
        <img
          alt="Waine Enterprises"
          src="/test5.png"
          style={
            Object {
              "height": "10em",
              "width": "auto",
            }
          }
        />
      </a>
    </div>
  </div>
</section>
`;

exports[`Sponsors renders correctly without any sponsor 1`] = `
<section
  id="sponsors"
>
  <div
    className="container"
  />
</section>
`;

exports[`Sponsors renders correctly without sponsor block titles 1`] = `
<section
  id="sponsors"
>
  <div
    className="container"
  >
    <h2>
      Our sponsors
    </h2>
    <div
      className="sponsor_block"
    >
      
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Tyrell Corp."
      >
        <img
          alt="Tyrell Corp."
          src="/test1.png"
          style={
            Object {
              "height": "15em",
              "width": "auto",
            }
          }
        />
      </a>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Black Mesa"
      >
        <img
          alt="Black Mesa"
          src="/test2.png"
          style={
            Object {
              "height": "15em",
              "width": "auto",
            }
          }
        />
      </a>
    </div>
    <div
      className="sponsor_block"
    >
      
      <a
        class="sponsor"
        href="https://fail.fail"
        title="IN Gen"
      >
        <img
          alt="IN Gen"
          src="/test3.png"
          style={
            Object {
              "height": "10em",
              "width": "auto",
            }
          }
        />
      </a>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Umbrella Corp"
      >
        <img
          alt="Umbrella Corp"
          src="/test4.png"
          style={
            Object {
              "height": "10em",
              "width": "auto",
            }
          }
        />
      </a>
      <a
        class="sponsor"
        href="https://fail.fail"
        title="Waine Enterprises"
      >
        <img
          alt="Waine Enterprises"
          src="/test5.png"
          style={
            Object {
              "height": "10em",
              "width": "auto",
            }
          }
        />
      </a>
    </div>
  </div>
</section>
`;
